### DISCUSSION TITLE
Infeasible Problem - what am I missing

### DISCUSSION THREAD

**Discussion by zsidanyi** [coin-or/pulp#393] [Q&A] [answered]
Hi,

i have a problem with the following variables:

_0 <= indicator_variable_0 <= 1 Integer
0 <= indicator_variable_1 <= 1 Integer
0 <= mem_pointer_0 <= 67108864 Integer
0 <= mem_pointer_1 <= 67108864 Integer
0 <= mem_pointer_2 <= 67108864 Integer
0 <= ram_budget Integer_

and constraints:

__C1:  mem_pointer_0 >= 196
_C2:  - mem_pointer_0 + ram_budget >= 0
_C3:  mem_pointer_1 >= 196
_C4:  - mem_pointer_1 + ram_budget >= 0
_C5:  mem_pointer_2 >= 1176
_C6:  - mem_pointer_2 + ram_budget >= 0
_C7:  - 67109060 indicator_variable_0 - mem_pointer_0 + mem_pointer_2 <= -196
_C8:  - 67110040 indicator_variable_0 - mem_pointer_0 + mem_pointer_2 >= -67108864
_C9:  - 67109060 indicator_variable_1 - mem_pointer_1 + mem_pointer_2 <= -196
_C10: - 67110040 indicator_variable_1 - mem_pointer_1 + mem_pointer_2 >= -67108864_

The _C7, _C8 and _C9, _C10 constraint pairs can be rewritten as:

**mem_pointer_0 - mem_pointer_2 not in [-1176, 196]
mem_pointer_1 - mem_pointer_2 not in [-1176, 196]**

based on the description of: 
https://www.researchgate.net/post/Is_it_possible_to_create_this_constraint_in_linear_programming (the answer from Adam N. Letchford):

> You can't have strict inequalities ("<" and ">") in mathematical programming. Perhaps you meant (x1 <= 2) or (x1 >= 8). This can be modelled provided three conditions hold:
> (i) You are allowed to include an additional binary variable.
> (ii) A lower bound L on the value of x1 is available.
> (iii) An upper bound U on the value of x1 is available.
> Under these conditions, it suffices to let y1 be a binary variable, taking the value 1 if and only if x1 >= 8, and add the constraints x1 <= 2 + (U-2) y1 and x1 >= L + (8-L) y1.

and considering that: 
- the upper boundary is 67108864
- the lower boundary is -67108864 
- indicator_variable_0 and indicator_variable_1 is defined as pulp.LpBinary

Based on these, I think a feasible solution would be:
x_0 = 196
x_1 = 196
x_2 = 2000

However I got an error at the end:
"Result - Problem proven infeasible"

I have PuLP version 2.3, and saved out an .mps file for the problem together with a log from cbc:
[problem.zip](https://github.com/coin-or/pulp/files/5836531/problem.zip)

What am I missing here?
Thank you for the answers in advance!

**Comment by pchtsp** [comment, marked as answer]
This can be or not related to your problem. You're using very large integer numbers (67108864) with integer variables and CBC usually gets confused when you do that. Try scaling your problem so it uses less significant digits.

**Reply by pchtsp** [reply]
Also check here for some general recommendations: https://coin-or.github.io/pulp/guides/how_to_debug.html#infeasible-problems

**Reply by zsidanyi** [reply]
Thank you for the answer and sorry for the late response. I tried out your recommendation and the problem was indeed the large numbers. 👍

**Reply by Fan9** [reply]
I also have this problem. If must use very large integer numbers  in my problem, what should i do? @pchtsp

**Reply by MattCremeens** [reply]
How to scale something like 3*x + 4*y = 10000000000000000?

**Reply by pchtsp** [reply]
You just need to scale the units of your x and y variables to something bigger. For example, if each unit of x currently represents one cell in your body, then change it to represent 10^14 cells.
